src/sqlancer/feldera/FelderaSchema.java

```diff
@@ -129,29 +114,144 @@ public FelderaExpression getRandomConstant(FelderaGlobalState globalState) {
 
             return FelderaConstant.getRandomConstant(globalState, this);
         }
+
+        public boolean isNumeric() {
+            return this.getPrimitiveType().isNumeric();
+        }
+
+        public FelderaCompositeDataType getElementType() {
+            return this.elementType;
+        }
+
+        public static FelderaCompositeDataType getBooleanType() {
```

Comment:
I don't understand what this class is supposed to be.
Maybe it's a type factory?

Reply:
this code should be in a separate factory class, which may also implement a singleton pattern.

Reply:
This is designed represents all datatypes in feldera (doesn't quite support map yet).
Similar to other implementations:
https://github.com/feldera/sqlancer/blob/23b8b437ddae168f3e60fab6cce1965bd8bea1a9/src/sqlancer/cockroachdb/CockroachDBSchema.java#L36